src/util/artifacts.js

```diff
@@ -1,25 +1,57 @@
 /* @flow */
 import {fs} from 'mz';
+import mkdirp from 'mkdirp';
+import promisify from 'es6-promisify';
 
 import {UsageError, isErrorWithCode} from '../errors';
 import {createLogger} from './logger';
 
 const log = createLogger(__filename);
-
+const defaultAsyncMkdirp = promisify(mkdirp);
 
 export async function prepareArtifactsDir(
-  artifactsDir: string
+  artifactsDir: string,
+  asyncMkdirp?: typeof defaultAsyncMkdirp = defaultAsyncMkdirp,
```

Comment:
To follow the convention we use everywhere else in web-ext, I'd like optional parameters to always be object properties. This solves a couple issues. First, it makes the calling code easier to read because you don't have to remember what all the positional arguments are. Secondly, it makes it easier to add additional options to this function in the future. The new signature would look like:
````js
type PrepareArtifactsDirOptions = {
  asyncMkdirp?: typeof defaultAsyncMkdirp,
}

export async function prepareArtifactsDir(
  artifactsDir: string,
  {asyncMkdirp = defaultAsyncMkdirp}: PrepareArtifactsDirOptions,
): Promise<string> {
  ...
}
````